### PR TITLE
Test --help and --version output in more cases

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

**Comment by medhefgo** [comment]
The version/help check script is basically the same. It imho makes more sense to unify it by passing an extra arg to it.

**Comment by keszybz** [comment]
> The version/help check script is basically the same. It imho makes more sense to unify it by passing an extra arg to it.

I considered that, but those checks change very rarely and it's not a big deal to have a little duplication. This way it's simple and easy to understand.

**Comment by yuwata** [comment]
From packit:
```
1041/1053 check-version-udevadm                                              FAIL              0.02s   exit status 4
>>> MALLOC_PERTURB_=104 /builddir/build/BUILD/systemd-251.rc1/tools/check-version.sh /builddir/build/BUILD/systemd-251.rc1/redhat-linux-build/udevadm v251.rc1-2.20220405163522435443.pr22980.145.gd13b69c073.fc37
â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€• âœ€  â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•
udevadm --version output does not contain v251.rc1-2.20220405163522435443.pr22980.145.gd13b69c073.fc37
â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•
```

File: tools/check-help.sh [hunk 1]
```diff
@@ -39,3 +39,9 @@ if ! ("$BINARY" --no-such-parameter 2>&1 1>/dev/null || :) | grep . >/dev/null;
     echo "$(basename "$BINARY") with an unknown parameter does not print to stderr"
     exit 4
 fi
+
+# --help and -h are equivalent
+if ! cmp <("$BINARY" -h) <("$BINARY" --help); then
```
Comment (mrc0mmand) [inline tools/check-help.sh]: Maybe use `diff` instead of `cmp`, so we see what's different? That should be especially helpful in CIs.

**Comment by keszybz** [comment]
Updated. The tests now output more info if there's a mismatch. I simplified the test for `--version` to just check for PROJECT_VERSION. It should be present in the output of both `udevadm` and other binaries.

**Review by yuwata — APPROVED** [review]
LGTM.

**Pull request merged**